### PR TITLE
Update operator --set help text

### PR DESCRIPTION
**Pull request by blhagadorn**
This PR was in the old operator location, https://github.com/istio/operator/pull/745 before it got moved over to istio/istio.  It fixes the issue raised here: https://github.com/istio/istio/issues/19999

[ ] Configuration Infrastructure
[ x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ x] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: operator/cmd/mesh/root.go [hunk 1]
```diff
@@ -24,9 +24,10 @@ import (
 )
 
 const (
-	SetFlagHelpStr = `Set a value in IstioOperator CustomResource. e.g. --set policy.enabled=true.
-Overrides the corresponding path value in the selected profile or passed through IstioOperator CR
-customization file`
+	SetFlagHelpStr = `Can be used to override a path in the IstioOperator API spec, e.g. to choose a profile
```
Comment (esnible) [inline operator/cmd/mesh/root.go]: Simplify grammar: `Override an IstioOperator value, e.g. ...`

**Review by esnible — CHANGES_REQUESTED** [review]
Also, please add `Example:` section to _istio/operator/cmd/mesh/manifest-apply.go_ and related commands that make use of `--set` showing a few common examples based on https://istio.io/docs/setup/install/istioctl/

```
istioctl manifest apply  # installs the default profile on the current Kubernetes cluster context
istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true # install the default profile with customizations
istioctl manifest apply --set installPackagePath=~/istio-releases/istio-1.4.3/install/kubernetes/operator/charts # install from external charts
istioctl manifest apply --set profile=demo # install a different profile.

File: operator/cmd/mesh/root.go (hunk 1 above)
Comment (blhagadorn) [reply operator/cmd/mesh/root.go]: That rolls of the tongue much better! Done.

Comment (blhagadorn) [reply operator/cmd/mesh/root.go]: I added the `Example:` to the cobraCommands for _manifest-apply.go_, however was unable to find any related commands that make use of `--set`.  Let me know if there is anything else I'm missing.

**Comment by blhagadorn** [comment]
Made necessary changes, not sure about that `gencheck_istio` job failing (where do I run `make gen`, in the base directory it throws a bunch of failures)?

**Comment by esnible** [comment]
I had huge amounts of problems with **gencheck_istio** on a PR that should have gone through on Friday.  I suspect something is wrong with it.

I was able to get my PR to pass using `BUILD_WITH_CONTAINER=1 make gen` which does the work inside a Docker container.  (I am on OSX, I am using Docker Desktop).  However, even after all of that, my PR didn't merge.  Let's not worry about gencheck right now.  Let's see if we can get another positive review of the change while the gencheck stuff gets sorted out elsewhere.

**Comment by howardjohn** [comment]
@esnible your PR not merging was a github bug, completely unrelated to gencheck. Just like in your PR, the go formatter has not run. This has been a part of Istio's CI forever and has not changed recently.

**Comment by blhagadorn** [comment]
Going to close this, sorry.

**Pull request closed without merging**